pkg/operator/options/options.go

```diff
@@ -57,66 +53,81 @@ func (s *nodeIdentitiesValue) String() string { return strings.Join(*s, ",") }
 
 type optionsKey struct{}
 
+// Options contains the configuration provided by the user.
+// Currently we do not support changing these after initialization in any way.
+// Even if we always get the updated value from the context/pointer, their copies that have been passed into external/vendored functions will not be updated.
+// And some of their defaults might even depends on neighbouring fields (e.g., APIServerName).
+// So, if one day we want to support dynamic configuration updates, consider creating the a new instance of Options and reinitialize the operator/controllers.
+//
+// If some fields need to be updated/refreshed and have their own (valid) cascading update procedure, consider moving it away to prevent confusion with the above assumption.
+// At that point, we should consider having a more clear distinction between user options and global variables.
 type Options struct {
-	ClusterName                    string
-	ClusterEndpoint                string // => APIServerName in bootstrap, except needs to be w/o https/port
-	VMMemoryOverheadPercent        float64
-	ClusterID                      string
+	// Target cluster information; might be use for both bootstrapping and ARM authentications
```

Comment:
The idea behind this is I try to categorize this by user-facing behavior (e.g., target cluster for the operations to occur, what will be on provisioned nodes) rather than implementation details.

Reply:
Some should be renamed. But let's put that in a different PR